.github/workflows/build.yaml

```diff
@@ -20,30 +20,68 @@ on:
     - cron: '0 3 * * *'
 
 jobs:
+
   build_and_push:
+
     name: Build ruby_${{ matrix.version }} images and push to GHCR
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        version: ['3_1', '3_1_2', '3_2', '3_2_0']
+        version:
+          - "['3', '1', '4']"
+          - "['3', '2', '2']"
     permissions:
       packages: write
+      
     steps:
+      
       - name: Login to GHCR
-        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a  # v2.1.0
+        uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
+        
       - uses: actions/checkout@v4
         with:
           ref: ${{ inputs.gitRef }}
           show-progress: false
-      - name: Build images (without pushing to registry)
-        if: ${{ inputs.noPushToRegistry }}
-        env:
-          DRY_RUN: "1"
-        run: ./build.sh ${{ matrix.version }}
-      - name: Build and push images
-        if: ${{ !inputs.noPushToRegistry }}
-        run: ./build.sh ${{ matrix.version }}
+      
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build base image and push
+        id: base
+        uses: docker/build-push-action@v5
+        with:
+          file: base.Dockerfile
+          context: .
+          platforms: linux/amd64,linux/arm64
+          build-args: |
+            RUBY_MAJOR=${{ fromJson(matrix.version)[0] }}.${{ fromJson(matrix.version)[1] }}
+            RUBY_VERSION=${{ join(fromJson(matrix.version), '.') }}
+          tags: |
+            gchr.io/${{ github.repository_owner }}/govuk-ruby-base:${{ fromJson(matrix.version)[0] }}.${{ fromJson(matrix.version)[1] }}
+            gchr.io/${{ github.repository_owner }}/govuk-ruby-base:${{ join(fromJson(matrix.version), '.') }}
+            gchr.io/${{ github.repository_owner }}/govuk-ruby-base:latest
```

Comment:
This is probably bad because the 3.1.x build might finish after the 3.2.x build ...